compiler/src/model/utils.ts

```diff
@@ -648,6 +635,34 @@ export function hoistRequestAnnotations (
       const docUrl = docIds.find(entry => entry[0] === value.trim())
       assert(jsDocs, docUrl != null, `The @doc_id '${value.trim()}' is not present in _doc_ids/table.csv`)
       endpoint.docUrl = docUrl[1]
+    } else if (tag === 'availability') {
+      // The @availability jsTag is different than most because it allows
+      // multiple values within the same docstring, hence needing to parse
+      // the values again in order to preserve multiple values.
+      const jsDocsMulti = parseJsDocTagsAllowDuplicates(jsDocs)
+      const availabilities = parseAvailabilityTags(jsDocs, jsDocsMulti.availability)
+
+      // Apply the availabilities to the Endpoint.
+      for (const availabilityName in availabilities) {
+        const availabilityValue = availabilities[availabilityName]
+        endpoint.availability[availabilityName] = availabilityValue
+
+        // Backfilling deprecated fields on an endpoint.
+        if (availabilityName === 'stack') {
+          if (availabilityValue.since !== undefined) {
+            endpoint.since = availabilityValue.since
+          }
+          if (availabilityValue.stability !== undefined) {
+            endpoint.stability = availabilityValue.stability
+          }
+          if (availabilityValue.visibility !== undefined) {
+            endpoint.visibility = availabilityValue.visibility
+          }
+          if (availabilityValue.featureFlag !== undefined) {
+            endpoint.featureFlag = availabilityValue.featureFlag
+          }
```

Comment:
```suggestion
          endpoint.since = availabilityValue?.since
          endpoint.stability = availabilityValue?.stability
          endpoint.visibility = availabilityValue?.visibility
          endpoint.featureFlag = availabilityValue?.featureFlag
```
optional chaining ftw :smile:

Reply:
This actually isn't the same operation, your suggestion unconditionally sets the value to `undefined` where there might already be a value defined on `endpoint.property`.

Reply:
ooh good catch. :star: